geoengine/workflow.py

```diff
@@ -599,7 +613,19 @@ def merge_tiles(tiles: List[xr.DataArray]) -> Optional[xr.DataArray]:
             if len(tiles) == 0:
                 return None
 
-            combined_tiles = xr.combine_by_coords(tiles)
+            # combine the bands of the tiles into multi-band tiles
+            tiles_by_tile_idx: Dict[Tuple[int, int], List[xr.DataArray]] = defaultdict(list)
+
+            for tile in tiles:
+                tile_idx = (tile.attrs['tile_idx_y'], tile.attrs['tile_idx_x'])
+                tiles_by_tile_idx[tile_idx].append(tile)
+
+            multi_band_tiles = []
+            for tile_idx, tiles_for_idx in tiles_by_tile_idx.items():
+                multi_band_tiles.append(xr.concat(tiles_for_idx, dim='band'))
```

Comment:
is it necessary to do multiple concats in advance to the combine?

Doesn't xarray put everything in place when the metadata is correct?

Reply:
or https://docs.xarray.dev/en/stable/generated/xarray.combine_nested.html#xarray.combine_nested ?

Reply:
the method seems to be doing exactly the same?

> Explicitly combine an N-dimensional grid of datasets into one by using a succession of concat and merge operations along each dimension of the grid.

I also use a succession of concat operations.

Reply:
I don't know, but why not use their API?

Reply:
sure, but how? How can I tell xarray to use the metadata (tile_idx) for determining which arrays to concatenate? The tile_idxs are deliberately not dimensions.

Reply:
I guess you don't? You have two dimensions, band and geo, and xarray can merge it based on these two?

Reply:
I would say remove it for now and if we need it someday, we'll bring it back.

Reply:
dba3b685f6aa7df546534f4d9b8c2319f51cb731